dotnet-sourcelink-shared/SourceLinkMap.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+// TODO: Use a source package once available (https://github.com/dotnet/sourcelink/issues/443)
```

Comment:
This looks to be waiting on this https://github.com/dotnet/sourcelink/issues/443 ?

Reply:
I don't think we need to wait for that work item to be done. I just haven't had time to finish this PR. Feel free to take it over.